test/index.test.ts

```diff
@@ -771,6 +771,463 @@ describe('Unit | Utility | changeset', () => {
+
+      expect(changeset.get('emails.0.funEmail')).toEqual('fun@email.com');
+      expect(changeset.changes).toEqual([{ key: 'emails.0.funEmail', value: 'fun@email.com' }]);
+      expect(changeset.get('emails').unwrap()).toEqual([
+        { primary: 'bob@email.com', funEmail: 'fun@email.com' }
+      ]);
+    });
+
+    it('can add new properties to new entries', () => {
+      const changeset = Changeset(initialData);
+
+      changeset.set('emails.1.funEmail', 'fun@email.com');
+      changeset.set('emails.1.primary', 'primary@email.com');
+
+      expect(changeset.get('emails.1.funEmail')).toEqual('fun@email.com');
+      expect(changeset.get('emails.1.primary')).toEqual('primary@email.com');
+      expect(changeset.get('emails').unwrap()).toEqual([
+        { primary: 'bob@email.com' },
+        { primary: 'primary@email.com', funEmail: 'fun@email.com' }
+      ]);
+      expect(changeset.changes).toEqual([
+        { key: 'emails.1.funEmail', value: 'fun@email.com' },
+        { key: 'emails.1.primary', value: 'primary@email.com' }
+      ]);
+    });
+
+    it('can add a new object all at once, and edit it', () => {
+      const changeset = Changeset(initialData);
+
+      changeset.set('emails.1', {
+        funEmail: 'fun@email.com',
+        primary: 'primary@email.com'
+      });
+
+      expect(changeset.get('emails.1.funEmail')).toEqual('fun@email.com');
+      expect(changeset.get('emails.1.primary')).toEqual('primary@email.com');
+      expect(changeset.get('emails').unwrap()).toEqual([
+        { primary: 'bob@email.com' },
+        { primary: 'primary@email.com', funEmail: 'fun@email.com' }
+      ]);
+      expect(changeset.changes).toEqual([
+        {
+          key: 'emails.1',
+          value: { funEmail: 'fun@email.com', primary: 'primary@email.com' }
+        }
+      ]);
+
+      changeset.set('emails.1.primary', 'primary2@email.com');
+
+      expect(changeset.get('emails.1.primary')).toEqual('primary2@email.com');
+      expect(changeset.changes).toEqual([
+        {
+          key: 'emails.1',
+          value: {
+            primary: 'primary2@email.com',
+            funEmail: 'fun@email.com'
+          }
+        }
+      ]);
+      expect(changeset.get('emails').unwrap()).toEqual([
+        { primary: 'bob@email.com' },
+        { primary: 'primary2@email.com', funEmail: 'fun@email.com' }
+      ]);
+    });
+
+    it('can edit a new object that was added after deleting an array entry', () => {
+      const changeset = Changeset({
+        emails: [
+          {
+            fun: 'fun0@email.com',
+            primary: 'primary0@email.com'
+          },
+          {
+            fun: 'fun1@email.com',
+            primary: 'primary1@email.com'
+          }
+        ]
+      });
+
+      changeset.set('emails.1', null);
+
+      expect(changeset.get('emails').unwrap()).toEqual([
+        {
+          fun: 'fun0@email.com',
+          primary: 'primary0@email.com'
+        },
+        null
+      ]);
```

Comment:
Does this test pass if you add these assertions (similar to the test below)?

```
expect(changeset.get('emails.0.fun')).toEqual('fun0@email.com');
expect(changeset.get('emails.0.primary')).toEqual('primary0@email.com');
```

Reply:
oh interesting. This is the diff with those:
![image](https://user-images.githubusercontent.com/199018/108858204-e5383e00-75b9-11eb-82bd-21062c56066c.png)

Reply:
I think this is a more general bug though, and why my work stuff _can't_ use the published validated changeset. 

_sometimes_ when accessing data via `get`, an object returns.
In _this_ case, the object is just totally incorrect. :thinking:

Reply:
Ok so same as the one below (was just checking if the `value` was the culprit).

Overall, this is a really great PR and I think we are really close.  I think we can focus on `.get` or `ObjectTreeNode` may need to be modified.  I can certainly jump in and see where the last few improvements need to be made.

Reply:
I'll take another stab, and see what happens :D

Reply:
fixed it -- wasn't so bad after all :D